<launch>
<param name="/use_sim_time" value="true"/>
 <node pkg="rosbag" type="play" name="loop_autol" args="--clock /home/polimi/robotics/src/bags/robotics2.bag -r 1" required="false" output="screen"></node>

<node pkg="second_project" name="wheel_o2t" type="odometry_to_tf" >
  <remap to="/ugv/odom" from="/input_odom"/>
  <param name="root_frame" value="odom" />
  <param name="child_frame" value="base_footprint" />
</node>

<!-- <node pkg="second_project" name="rslidar_to_odom" type="swap_tf" >
  <param name="dest_tf" value="wheel_odom" />
</node> -->

<node pkg="slam_toolbox" type="async_slam_toolbox_node" name="slam_toolbox" output="screen">
  <rosparam command="load" file="$(find second_project)/config/mapper_params_online_async.yaml" />
</node>


<!-- <node pkg="pointcloud_to_laserscan" name="pc2lsodomframe" type="pointcloud_to_laserscan_node" >
  <remap to="/pointcloud_remapped" from="/cloud_in"/>
  <remap to="/scanodomframe" from="/scan"/>
  <param name="max_height" value="0.5" />
</node> -->

<node pkg="pointcloud_to_laserscan" name="pc2ls" type="pointcloud_to_laserscan_node" >
  <remap to="/ugv/rslidar_points" from="/cloud_in"/>
  <param name="max_height" value="0.5" /> 
  <param name="range_max" value="5.5" /> 
  
</node>

<node type="rviz" name="rviz" pkg="rviz"  args="-d  $(find second_project)/lidar.rviz" ></node> 
</launch>
